### PR TITLE
@mzikherman - Use reaction-force gene component

### PR DESCRIPTION
**Pull request by broskoski**
This makes the gene page use the reaction-force components for the main filter section instead of the backbone stuff that is currently there. Still some work to do with styling!

screenshot 2017-05-30 18 15 59

### CODE REVIEW

File: desktop/apps/gene/client.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import React from 'react'
+import ReactDOM from 'react-dom'
+import * as Relay from 'react-relay'
+import { data as sd } from 'sharify'
+
+import { artsyNetworkLayer } from '@artsy/reaction-force/dist/relay/config'
+import components from '@artsy/reaction-force/dist/components/index'
+import * as Artsy from '@artsy/reaction-force/dist/components/artsy'
+import GeneQueryConfig from '@artsy/reaction-force/dist/relay/queries/gene'
```
Comment (alloy) [inline desktop/apps/gene/client.js]: We should maybe export these often needed modules from the root index file?

File: desktop/apps/gene/client.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+function init () {
+  Relay.injectNetworkLayer(artsyNetworkLayer(sd.CURRENT_USER))
+  ReactDOM.render(
+    <Artsy.ContextProvider currentUser={sd.CURRENT_USER}>
+      <Relay.RootContainer
+        Component={components.Gene}
+        route={new GeneQueryConfig({ geneID: sd.GENE.id })}
+      />
+    </Artsy.ContextProvider>, document.getElementById('gene-filter')
+  )
```
Comment (alloy) [inline desktop/apps/gene/client.js]: If this boilerplate starts to become a pattern, then let’s export a generic function from reaction-force that does all of this.

File: desktop/apps/gene/client.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  // Load related artists
+  let gene = new Gene(sd.GENE)
```
Comment (alloy) [inline desktop/apps/gene/client.js]: `const` ?

File: desktop/apps/gene/client.js (hunk 1 above)
Comment (broskoski) [inline desktop/apps/gene/client.js]: Yeah, agreed.

File: desktop/apps/gene/client.js (hunk 2 above)
Comment (broskoski) [inline desktop/apps/gene/client.js]: Agreed as well.

File: desktop/apps/gene/client.js (hunk 3 above)
Comment (broskoski) [inline desktop/apps/gene/client.js]: Agreed, agreed, a thousand times agreed

File: desktop/apps/gene/client.js [hunk 4]
```diff
@@ -3,15 +3,9 @@ import $ from 'jquery'
...
-import ReactDOM from 'react-dom'
-import * as Relay from 'react-relay'
 import { data as sd } from 'sharify'
-
-import { artsyNetworkLayer } from '@artsy/reaction-force/dist/relay/config'
-import components from '@artsy/reaction-force/dist/components/index'
-import * as Artsy from '@artsy/reaction-force/dist/components/artsy'
-import GeneQueryConfig from '@artsy/reaction-force/dist/relay/queries/gene'
+import components, { init } from '@artsy/reaction-force/dist/components/index'
+import configs from '@artsy/reaction-force/dist/relay/index'
```
Comment (alloy) [inline desktop/apps/gene/client.js]: Nit: No need to include the `index` filename.

**Comment by broskoski** [comment]
It's ALIVE! Ready when you are @mzikherman 

gene-reaction-force

File: desktop/apps/gene/client.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+const relatedArtistsTemplate = (args) => {
+  return require('./templates/related_artists.jade')(args)
+}
+
+function setupGenePage () {
+  // Init GenePage component
+  const urlParams = qs.parse(location.search.replace(/^\?/, ''))
+  const options = Object.assign({}, urlParams, { geneID: sd.GENE.id })
+
+  console.log('options', options)
```
Comment (mzikherman) [inline desktop/apps/gene/client.js]: 🌴

**Comment by mzikherman** [comment]
stray log but aw yea!

File: desktop/apps/loyalty/index.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
 const express = require('express')
 
 const app = module.exports = express()
-app.use('/loyalty', require('@artsy/reaction-force').default.apps.loyalty)
+app.use('/loyalty', require('@artsy/reaction-force/dist/apps/loyalty/server').default)
```
Comment (l2succes) [inline desktop/apps/loyalty/index.js]: @broskoski Why did you need to make that change btw, did it break for some reason?

Comment (broskoski) [inline desktop/apps/loyalty/index.js]: It did, for the same reason we have to require components from the `dist` folder on the client-side (but for the opposite reason). All the server stuff (`artsy-passport`, etc) breaks the client-side requires, and now `history` will break the server-side requires (if we don't require from the direct path).

I still maintain that reaction-force is not the place to build entire apps but rather the components that will be used to build an app. The gluing should be done outside reaction-force IMO.
